sklearn/covariance/elliptic_envelope.py

```diff
@@ -97,79 +95,92 @@ def __init__(self, store_precision=True, assume_centered=False,
+            in 0.22.
+
         Returns
         -------
+
         decision : array-like, shape (n_samples, )
             Decision function of the samples.
-            It is equal to the Mahalanobis distances if `raw_values`
-            is True. By default (``raw_values=False``), it is equal
-            to the cubic root of the shifted Mahalanobis distances.
-            In that case, the threshold for being an outlier is 0, which
-            ensures a compatibility with other outlier detection tools
-            such as the One-Class SVM.
+            It is equal to the shifted Mahalanobis distances.
+            The threshold for being an outlier is 0, which ensures a
+            compatibility with other outlier detection algorithms.
 
         """
-        check_is_fitted(self, 'threshold_')
-        X = check_array(X)
-        mahal_dist = self.mahalanobis(X)
-        if raw_values:
-            decision = mahal_dist
-        else:
-            transformed_mahal_dist = mahal_dist ** 0.33
-            decision = self.threshold_ ** 0.33 - transformed_mahal_dist
+        negative_mahal_dist = self.score_samples(X)
 
-        return decision
+        # raw_values deprecation:
+        if raw_values is not None:
+            warnings.warn("raw_values parameter is deprecated in 0.20 and will"
+                          " be removed in 0.22.", DeprecationWarning)
 
-    def predict(self, X):
-        """Outlyingness of observations in X according to the fitted model.
+            if not raw_values:
+                return (-self.offset_) ** 0.33 - (-negative_mahal_dist) ** 0.33
+
+        return negative_mahal_dist - self.offset_
+
+    def score_samples(self, X):
+        """Compute the Mahalanobis distances.
 
         Parameters
         ----------
-        X : array-like, shape = (n_samples, n_features)
+        X : array-like, shape (n_samples, n_features)
 
         Returns
         -------
-        is_outliers : array, shape = (n_samples, ), dtype = bool
-            For each observation, tells whether or not it should be considered
-            as an outlier according to the fitted model.
+        negative_mahal_distances : array-like, shape (n_samples, )
+            Opposite of the Mahalanobis distances.
+        """
+        check_is_fitted(self, 'offset_')
+        X = check_array(X)
+        return -self.mahalanobis(X)
+
+    def predict(self, X):
+        """
+        Predict the labels (1 inlier, -1 outlier) of X according to the
+        fitted model.
 
-        threshold : float,
-            The values of the less outlying point's decision function.
+        Parameters
+        ----------
+        X : array-like, shape = (n_samples, n_features)
 
+        Returns
+        -------
+        is_inlier : array, shape (n_samples,)
+            Returns -1 for anomalies/outliers and +1 for inliers.
         """
-        check_is_fitted(self, 'threshold_')
+        check_is_fitted(self, 'offset_')
```

Comment:
This doesn't need repeating if done in decision_function.

Should this definition of predict be provided by a mixin?

Reply:
@jnothman do you mean a mixin for outlier detection estimators? I was thinking that all outlier detection estimators should have a `fit_predict` like clustering estimators but they can't all have a `predict` unless we exclude `LocalOutlierFactor` which has a private `predict`.

Reply:
Fair enough. We can make the presence of predict conditional on the presence of `decision_function`. It's a bit ugly, but possible with a descriptor/property...

Reply:
I propose to create such a mixin in a separate PR, as this one is already a bit heavy

Reply:
I will take care of this once this PR is merged.